irma/ftp/sftp.py

```diff
@@ -53,7 +53,12 @@ def _connect(self):
             self._conn.window_size = pow(2, 27)
             self._conn.packetizer.REKEY_BYTES = pow(2, 32)
             self._conn.packetizer.REKEY_PACKETS = pow(2, 32)
-            self._conn.connect(username=self._user, password=self._passwd)
+            if self._auth == 'key':
+                pkey = RSAKey.from_private_key_file('/home/irma/.ssh/id_rsa')
```

Comment:
good point. I want to keep sftp behaviour as close as FTP as possible. Meaning virtual users chrooted with password auth.
Maybe a config with PasswordAuthentication disabled for all users and enabled only for sftp group (that could only do sftp not ssh).
Anyway will try to find a good place for the app private key and review your PR.